File: codeforces/gym/brazil-2024/k.py
n = int(input())
lista = list(map(int, input().split()))
target = sum(lista) // 2

dp = [0] * 10010
for i in range(n): 
	if lista[i] > target: continue
	for k in range(target, 0, -1):
		if dp[k] != 0 and (k + lista[i]) <= target:
			dp[k + lista[i]] = dp[k].copy()
			dp[k + lista[i]][i] = 1
			
	dp[lista[i]] = {}
	dp[lista[i]][i] = 1
		
primeiro = [] 
segundo = []

if dp[target] == 0 or n == 1 or sum(lista) % 2:
	print(-1)
else:
	for i in range(0, n):
		if (dp[target].get(i) == 1):
			primeiro.append(lista[i])
		else:
			segundo.append(lista[i]) 

	soma = 0
	ind1 = 0
	ind2 = 0
	for i in range(0, n):
		if soma > 0:
			print(primeiro[ind1], end=" ")
			soma -= primeiro[ind1]
			ind1 += 1
		else:
			print(segundo[ind2], end=" ")
			soma += segundo[ind2]
			ind2 += 1
	print()

File: codeforces/gym/brazil-2024/k.cpp
#include<bits/stdc++.h>

using namespace std;

int valores[105];
bitset<105> dp[10010];


signed main() {
	int n; cin>>n;

	int target = 0;
	for (int i = 0; i < n; i++) {
		cin>>valores[i];
		target += valores[i];
	}

	if (n == 1 || target%2) {
		cout<<-1<<endl;
		return 0;
	}

	for (int i = 0; i < target; i++) {
		dp[i].reset();
	}

	target /= 2;
	
	for (int i = 0; i < n && dp[target].count() == 0; i++) {
		for (int k = target; k >= 1; k--) {
			if (dp[k].count() > 0) {
				cout<<i<<endl;
				bitset<105> copy(dp[k]);
				dp[k+valores[i]] = copy;
				dp[k+valores[i]].set(i);

				if (k+valores[i] == target) {
					break;
				}	
			}
		}
		dp[valores[i]].set(i);
	}

	if (dp[target].count() == 0) {
		cout<<-1<<endl;
		return 0;
	}

	vector<int> primeiro;
	vector<int> segundo;
	for (int i = 0; i < n; i++) {
		if (dp[target].test(i)) {
			primeiro.push_back(valores[i]);
		} else {
			segundo.push_back(valores[i]);
		}
	}

	int ind1 = 0;
	int ind2 = 0;
	int soma = 0;

	for (int i = 0; i < n; i++) {
		if (soma >= 0) {
			soma -= primeiro[ind1];
			cout<<primeiro[ind1++]<<" ";
		} else {
			soma += segundo[ind2];
			cout<<segundo[ind2++]<<" "; 
		}
	}

	cout<<endl;
	return 0;
}